File: app/backend/java/ride-service/src/main/java/com/srbms/rideservice/entity/Ride.java
package com.srbms.rideservice.entity;

import jakarta.persistence.*;

@Entity
@Table(name = "ride")
public class Ride {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ride_id")
    private Integer rideId;

    @Column(name = "user_id", nullable = false)
    private Integer userId;

    @Column(name = "vehicle_id", nullable = false)
    private Integer vehicleId;

    @Column(name = "source", nullable = false)
    private String source;

    @Column(name = "destination", nullable = false)
    private String destination;

    @Column(name = "status")
    private Integer status; // 1 = completed, 2 = inprogress, 0 = requested, 3 = accepted

    @Column(name = "fare")
    private Double fare;

    @Column(name = "created_at", insertable = false, updatable = false)
    private java.time.LocalDateTime createdAt;

    public Ride() {}

    public Ride(Integer userId, Integer vehicleId, String source, String destination, Integer status) {
        this.userId = userId;
        this.vehicleId = vehicleId;
        this.source = source;
        this.destination = destination;
        this.status = status;
    }

    public Ride(Integer userId, Integer vehicleId, String source, String destination, Integer status, Double fare) {
        this.userId = userId;
        this.vehicleId = vehicleId;
        this.source = source;
        this.destination = destination;
        this.status = status;
        this.fare = fare;
    }

    public Integer getRideId() {
        return rideId;
    }

    public void setRideId(Integer rideId) {
        this.rideId = rideId;
    }

    public Integer getUserId() {
        return userId;
    }

    public void setUserId(Integer userId) {
        this.userId = userId;
    }

    public Integer getVehicleId() {
        return vehicleId;
    }

    public void setVehicleId(Integer vehicleId) {
        this.vehicleId = vehicleId;
    }

    public String getSource() {
        return source;
    }

    public void setSource(String source) {
        this.source = source;
    }

    public String getDestination() {
        return destination;
    }

    public void setDestination(String destination) {
        this.destination = destination;
    }

    public Integer getStatus() {
        return status;
    }

    public void setStatus(Integer status) {
        this.status = status;
    }

    public java.time.LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(java.time.LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public Double getFare() {
        return fare;
    }

    public void setFare(Double fare) {
        this.fare = fare;
    }
}

File: app/backend/.net/SmartRide.RideService/Models/Entities/Ride.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SmartRide.RideService.Models.Entities;

[Table("ride")]
public class Ride
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    [Column("ride_id")]
    public int RideId { get; set; }

    [Column("user_id")]
    public int UserId { get; set; }

    [Column("vehicle_id")]
    public int VehicleId { get; set; }

    [Column("source")]
    public string Source { get; set; } = string.Empty;

    [Column("destination")]
    public string Destination { get; set; } = string.Empty;

    [Column("status")]
    public int Status { get; set; } // 0 = Requested, 1 = Completed, 2 = InProgress, 3 = Accepted

    [Column("fare")]
    public decimal? Fare { get; set; }

    [Column("created_at")]
    public DateTime? CreatedAt { get; set; }

    public Ride() { }

    public Ride(int userId, int vehicleId, string source, string destination, int status, decimal? fare = null)
    {
        UserId = userId;
        VehicleId = vehicleId;
        Source = source;
        Destination = destination;
        Status = status;
        Fare = fare;
    }
}
